lib/cgi/session.rb

```diff
@@ -210,15 +210,19 @@ def create_new_id
     # suffix:: the prefix to add to the session id when generating
     #          the filename for this session's FileStore file.
     #          Defaults to the empty string.
+    # digest:: the digest algorithm to hash the session id when
+    #          generating the filename for this session's FileStore
+    #          file.  Defaults to "MD5".
     def new_store_file(option={}) # :nodoc:
       dir = option['tmpdir'] || Dir::tmpdir
       prefix = option['prefix']
       suffix = option['suffix']
-      require 'digest/md5'
-      md5 = Digest::MD5.hexdigest(session_id)[0,16]
+      algorithm = option['digest'] || 'MD5'
+      require 'digest'
+      digest = Digest(algorithm).hexdigest(session_id)[0,16]
```

Comment:
Why do we limit the size to 16 characters? This could be a security issue.

Reply:
It should be another option.